m4/with_libical.m4

```diff
@@ -72,10 +72,12 @@ AC_DEFUN([PBS_AC_WITH_LIBICAL],
   AC_SUBST(libical_inc)
   AC_SUBST(libical_lib)
   AC_DEFINE([LIBICAL], [], [Defined when libical is available])
-  PKG_CHECK_MODULES([libical_api2],
-    [libical >= 2],
-    [AC_DEFINE([LIBICAL_API2], [], [Defined when libical version >= 2])],
-    [echo "libical version 2 is not available"]
-    )
+  AS_IF([test "$libical_dir" = "/usr"],
+    [PKG_CHECK_MODULES([libical_api2],
+      [libical >= 2],
+      [AC_DEFINE([LIBICAL_API2], [], [Defined when libical version >= 2])],
+      [echo "libical version 2 is not available"]
+    )]
+  )
```

Comment:
What happens if $libical_dir is something other than /usr but contains libical version 2? This solution only works when $libical_dir is /usr.

Reply:
For LIBICAL_API2 macro , the code was originally checking if libical version 2 is installed in the machine (using  PKG_CHECK_MODULES), which it still does. I don't think the code was originally handling the $libical_dir != /usr case either. These changes will ensure the above mentioned case doesn't cause any problems.

Reply:
@saimanikant - we cannot be sure that libical is always installed in /usr. In OSS, we cannot have that restriction.
Also, in the PR summary - please explain why are you checking for /usr.

Reply:
@prakashcv13 the value of $libical_dir comes from here:
https://github.com/PBSPro/pbspro/blob/fac65ee53a49189ea2ee35cdf8968538cd0118d7/m4/with_libical.m4#L46-L49

Anyways, I'll be pushing new changes for version 2 check for both the above cases.